CMakeLists.txt

```diff
@@ -303,7 +303,7 @@ if(BUILD_TESTS)
       raft_test ${CMAKE_CURRENT_SOURCE_DIR}/src/consensus/aft/test/main.cpp
       ${CMAKE_CURRENT_SOURCE_DIR}/src/consensus/aft/test/view_history.cpp
     )
-    target_link_libraries(raft_test PRIVATE ccfcrypto.host)
+    target_link_libraries(raft_test PRIVATE ccfcrypto.host http_parser.host)
```

Comment:
How does the raft_test now depend on the http parser?

Reply:
Via `make_request` in `ByzantineIdentityTracker`; the `Aft` class is now able to make HTTP calls. If there's a better way to do this, I'm happy to change it.

Reply:
I think we should avoid this dependency with another layer of abstraction. Either by templating (the tests take a `StubByzantineIdentityTracker` which doesn't do HTTP calls, which we already do for the ledger type for instance), or inheritance (`Aft` talks to a `ByzantineIdentityTracker` that has some stub virtual calls where it would send a HTTP request, and the actual instance is a `ByzantineIdentityTrackerWithDescriptiveName` that overrides those and needs HTTP-parser linkage). Either approach allows stub implementations that would make this unit testable - ie rather than creating HTTP requests, we just create an in-memory stack of requests, and the test can assert that the required calls were made.

Reply:
That's a good suggestion, I'll add a stub tracker!